griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,335 @@
+            ): [str]
+        })
+    })
+    def download_files(self, params: dict) -> ListArtifact | ErrorArtifact:
+        values = params["values"]
+        downloaded_files = []
+    
+        service = self._build_client(self.LIST_FILES_SCOPES)
+    
+        for path in values["paths"]:
+            try:
+                file_id = self._path_to_file_id(service, path)
+                if file_id:
+                    file_info = service.files().get(fileId=file_id).execute()
+                    mime_type = file_info["mimeType"]
+    
+                    if mime_type in self.GOOGLE_EXPORT_MIME_MAPPING:
+                        export_mime = self.GOOGLE_EXPORT_MIME_MAPPING[mime_type]
+                        request = service.files().export_media(fileId=file_id, mimeType=export_mime)
+                    else:
+                        request = service.files().get_media(fileId=file_id)
+    
+                    downloaded_file = request.execute()
+                    logging.info(f"File '{path}' successfully downloaded.")
+                    downloaded_files.append(BlobArtifact(downloaded_file))
+                else:
+                    logging.error(f"Could not find file: {path}")
+                    downloaded_files.append(
+                        ErrorArtifact(f"Could not find file: {path}")
+                    )
+    
+            except HttpError as e:
+                logging.error(e)
+                downloaded_files.append(
+                    ErrorArtifact(f"error downloading file '{path}' from Google Drive: {e}")
+                )
+            except MalformedError:
+                logging.error("MalformedError occurred")
+                downloaded_files.append(
+                    ErrorArtifact(f"error downloading file '{path}' from Google Drive due to malformed credentials")
+                )
+    
+        return ListArtifact(downloaded_files)
+
+    @activity(
+        config={
+            "description": "Can search for files on Google Drive based on name or content",
+            "schema": Schema(
+                {
+                    Literal(
+                        "search_mode",
+                        description="File search mode. Use 'name' to search in file name or "
+                                    "'content' to search in file content",
+                    ): Or("name", "content"),
+                    Literal(
+                        "search_query",
+                        description="Query to search for. If search_mode is 'name', it's the file name. If 'content', "
+                        "it's the text within files.",
+                    ): str,
+                    Literal(
+                        "folder_path",
+                        description="Path of the Google Drive folder (like 'MainFolder/Subfolder1/Subfolder2') "
+                        "where the search should be performed. Specify 'root' to search in the "
```

Comment:
Once it's made optional, you don't need to prompt the LLM to provide "root" as a value.

Reply:
Done!